### PR TITLE
OCPBUGS-23925: Only show VPAs that are associated with the selected namespace

### PR DESCRIPTION
**Pull request by sg00dwin**
https://issues.redhat.com/browse/OCPBUGS-23925

### CODE REVIEW

File: frontend/packages/topology/src/components/workload/vpa-tab-section.tsx [hunk 1]
```diff
@@ -51,11 +51,14 @@ export const useVpaSideBarTabSection: DetailsTabSectionExtensionHook = (element:
...
   const resource = getResource(element);
+  const { namespace } = resource.metadata;
+  const vpasForNamespace = vpas.filter((vpa) => vpa.metadata.namespace === namespace);
+
   const verticalPodAutoscaler = getVerticalPodAutoscalerForResource(vpas, resource);
 
   const section = verticalPodAutoscaler ? (
     <TopologySideBarTabSection>
-      <VPATabSection vpas={vpas} />
+      <VPATabSection vpas={vpasForNamespace} />
```
Comment (rhamilto) [inline frontend/packages/topology/src/components/workload/vpa-tab-section.tsx:61]:
```suggestion
      <VPATabSection vpas={verticalPodAutoscaler} />
```
is what you want, no?  On the Deployment details page, we assume there is only one VPA.  I would think we want the same here.  If not, then I think we need to revisit the Deployment details page as well.

**Comment by cyril-ui-developer** [comment]
@sg00dwin I'm trying to understand what the fix is for. Could you pls explain more or add more description to the ticket. I reproduced the following by creating a deployment and vpa in the same ns, but other vpa's are displayed too. Is this right?

Screenshot 2023-12-07 at 3 49 54 PM

**Comment by rhamilto** [comment]
/lgtm

Tagging so this gets in before branching.

**Pull request merged**